src/cbsodatav4_kcvanderlinden/cbsodatav4.py

```diff
@@ -2,142 +2,188 @@
+        The URL of the data you want to retrieve.
+    Returns
+    -------
+    list
+        A list of dictionaries containing the retrieved data.
+    
+    Examples
+    --------
+    >>> cbsConnect('https://datasets.cbs.nl/odata/v1/CBS/$metadata')
     '''
     session = requests.Session()
     retries = Retry(total=5, backoff_factor=1, status_forcelist=[ 500, 502, 503, 504 ])
     session.mount('https://', HTTPAdapter(max_retries=retries))
     retry_count = 0
-    while retry_count < 5:
+    time_out_time = 0
+    while True:
         try:
             response = session.get(target_url)
             if response.status_code == 200:
-                return response.json()['value']
+                data = response.json()  # Load the JSON data directly from the response object
+                return data
             else:
                 raise requests.exceptions.RequestException(f"HTTP {response.status_code}")
         except requests.exceptions.ChunkedEncodingError:
-            print(f"ChunkedEncodingError occurred. Retrying...")
+            sys.stdout.write(f"ChunkedEncodingError occurred (data load incomplete). Retrying...")
             retry_count += 1
+            time_out_time += random.randrange(1, 10)/10 # dynamic time-out after unstable connection with host.
+            sleep(time_out_time)
             if retry_count == 5:
                 raise requests.exceptions.ChunkedEncodingError("Max retries exceeded")
+        except json.JSONDecodeError as e:
+            sys.stdout.write(f"JSON decode error: {e}. Retrying...")
 
-def getData(target_url:str, tableLength:int=0):
+def getData(target_url:str, return_data:bool = False):
     '''
     Get the data from the target URL.
-    target_url: str, the URL to get the data from
-    tableLength: int, the number of rows in the table
-    '''       
-    if tableLength > 100000:
-        queryAmount = int(math.ceil(tableLength / 100000))
-        queryJobs = [f"{target_url}?$skip={i*100000}" for i in range(queryAmount)]
-        
-        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
-            futures = [executor.submit(cbsConnect, job) for job in queryJobs]
-            concurrent.futures.wait(futures)
-            listOfListOfDicts = [future.result() for future in futures]
-            listOfDicts = [item for sublist in listOfListOfDicts for item in sublist]
-            df = pd.DataFrame(listOfDicts)
-    else:
-        r = requests.get(target_url).json()
-        df = pd.DataFrame(r['value'])
+
+    Parameters
+    ----------
+    target_url : str
+        The URL to retrieve data from.
+    return_data : bool, optional
+        If True, return the retrieved data directly. Otherwise, return None (default is False).
     
-    return df
+    Returns
+    -------
+    dict or None
+        A dictionary containing the retrieved data if return_data=True; otherwise, None.
+    '''
+    hex_target_url = ''.join(format(ord(c), 'x') for c in target_url)[-20:]
```

Comment:
is dit ook een statisch iets, die normaals gesproken niet aangepast hoeft te worden ([-20:])?

Reply:
is een keuze om de hex qua lengte van de filenaam te minimaliseren.